### PR TITLE
Some more helpful messages for anal.verbose ##anal

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

File: libr/core/canal.c [hunk 1]
```diff
@@ -1849,6 +1849,9 @@ R_API int r_core_anal_fcn(RCore *core, ut64 at, ut64 from, int reftype, int dept
 
 	if (core->io->va) {
 		if (!r_io_is_valid_offset (core->io, at, !core->anal->opt.noncode)) {
+			if (core->anal->verbose) {
```
Comment (XVilka) [inline libr/core/canal.c]: Probably we need a macro for that.

**Comment by radare** [comment]
We need to overengineer eprintf to make it work on cutter and r2pipe

Ideas?

**Comment by xarkes** [comment]
I personally don't like `eprintf` and I believe it should be used only as a shortcut for local debugging.
I think it would be better to use `r_log` or similar API.

**Comment by ret2libc** [comment]
Yes indeed we have r_log and some nice R_LOG macros as well. @radare we should use those IMHO, or build something on top of that to enable/disable logs "by module". We already have log.level, log.file, we could something like `log.modules` and if you set it to `log.modules=analysis,core,rbin.elf`

**Comment by radare** [comment]
but how can we make rlog output accessible from r2pipe for example? addinga c omand to collect all that buffer? eprintf use to be "streamy" not buffered like cons, so we maybe need a separate channel to do that

**Pull request merged**